### PR TITLE
dev/benchmark: add sysbench in k8s benchmark report

### PR DESCRIPTION
**Pull request by cofyc**
### What is changed, added or deleted? 

Add TiDB sysbench benchmark report in k8s (gke).

### Which version does your change affect? 

dev

https://github.com/cofyc/docs-cn/blob/sysbench-in-k8s/dev/benchmark/sysbench-in-k8s.md

### CODE REVIEW

File: dev/benchmark/sysbench-in-k8s.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+TiKV:
+
+```toml
+log-level = "error"
+[server]
+status-addr = "0.0.0.0:20180"
+grpc-concurrency = 6
+[readpool.storage]
```
Comment (weekface) [inline dev/benchmark/sysbench-in-k8s.md]: block-cache 在 tidb 3.0 的配置文件格式不是这个了吧？

File: dev/benchmark/sysbench-in-k8s.md [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+在单可用区测试中，我们选择了如下型号机器：
+
+| 组件     | 实例类型       | 数量  |
+| ----     | ----------     | ----- |
+| PD       | n1-standard-4  | 3     |
+| TiKV     | c2-standard-16 | 3     |
+| TiDB     | c2-standard-16 | 3     |
+| Sysbench | c2-standard-30 | 1     |
+
+在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如何机器型号进行测试:
```
Comment (weekface) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如下机器型号进行测试:
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+
+QPS 对比:
+
+![Service vs Load Balancer](/media/sysbench-in-k8s/service-vs-load-balancer-qps.svg)
+
+Latency 对比:
+
+![Service vs Load Balancer](/media/sysbench-in-k8s/service-vs-load-balancer-latency.svg)
+
+从图中可以看到 Ubuntu 系统在单纯的 Point Select 测试是中表现比 COS 系统要好。
```
Comment (weekface) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
从图中可以看到 K8S Service 在单纯的 Point Select 测试是中表现比 GCP LoadBalancer 要好。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+## 结语
+
+此次测试主要将典型共有云部署 Kubernetes 运行 TiDB 集群的几种场景使用 sysbench 做了测试，了解不同因素可能带来的影响。从整体看，主要有以下几点：
+
+- VPC-Native 模式下 Host 网络性能略好于 Pod 网络 (~7%，以 QPS 差异估算，下同)
+- GCP 的 Ubuntu 系统单纯的读测试中性能略好于 COS (~9%)
+- 使用 Load Balancer 在集群外访问，会略损失性能 (~5%)
+- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响 (30% ~ 6%，随并发数增加而下降)
+
+但要注意的是，这些因素可能随着时间变化，不同公有云下的表现可能会略有不同。同时，sysbench 测试用例并不能代表真实应用场景，在做选择前需做实际生产测试，并综合不通选择背后的代价（操作系统差异、Host 网络的限制等）。
```
Comment (weekface) [inline dev/benchmark/sysbench-in-k8s.md]: “并综合不通选择背后的代价”？

Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: fixed

File: dev/benchmark/sysbench-in-k8s.md (hunk 1 above)
Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: 是的，不过这些配置在 3.0 继续兼容。

File: dev/benchmark/sysbench-in-k8s.md [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+---
+
+# TiDB in Kubernetes Sysbench 性能测试
+
+随着 [TiDB Operator GA 发布](https://pingcap.com/blog-cn/tidb-operator-1.0-ga/)，越来越多用户开始使用 TiDB Operator 在 Kubernetes 中部署管理 TiDB 集群。在本次测试中，我们选择 GKE 平台做了一次深入、全方位的测试，方便大家了解 TiDB 在 Kubernetes 中性能影响因素。
+
+## 目的
+
+- 测试典型公有云平台上 TiDB 性能数据
+- 测试公有云平台磁盘、网络、CPU 以及不同 Pod 网络下 对 TiDB 性能影响
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
- 测试公有云平台磁盘、网络、CPU 以及不同 Pod 网络下对 TiDB 性能的影响
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+
+## 环境
+
+### 版本与配置
+
+本次测试统一使用 TiDB v3.0.1 版本进行测试。
+
+TiDB Operator 使用 v1.0.0 版本。
+
+PD, TiDB 和 TiKV 实例数均为 3 个。各组件分别作了如下配置，未配置部分使用默认值。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
PD、TiDB 和 TiKV 实例数均为 3 个。各组件分别作了如下配置，未配置部分使用默认值。
```

Comment (lilin90) [reply dev/benchmark/sysbench-in-k8s.md]: 这里的“实例”指的是 Pod 吗？

File: dev/benchmark/sysbench-in-k8s.md [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+在单可用区测试中，我们选择了如下型号机器：
+
+| 组件     | 实例类型       | 数量  |
+| ----     | ----------     | ----- |
+| PD       | n1-standard-4  | 3     |
+| TiKV     | c2-standard-16 | 3     |
+| TiDB     | c2-standard-16 | 3     |
+| Sysbench | c2-standard-30 | 1     |
+
+在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 没有 region 有 3 个可用区同时提供 c2 机器，我们选择了如下机器型号进行测试:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
在多可用区相对单可用区性能对比中，因为测试时 (2019.08) GCP 的 Region 均没有 3 个可用区可同时提供 c2 机器，我们选择了如下机器型号进行测试：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+| PD       | n1-standard-4  | 3     |
+| TiKV     | n1-standard-16 | 3     |
+| TiDB     | n1-standard-16 | 3     |
+| Sysbench | n1-standard-16 | 3     |
+
+在高并发读测试中，压测端 sysbench 对 CPU 需求较高，需选择多核较高配置型号，避免压测端成为瓶颈。
+
+#### 磁盘
+
+GKE 当前的 NVMe 磁盘还在 Alpha 阶段，使用需特殊申请，不具备普遍意义。测试中，本地 SSD 盘统一使用 iSCSI 接口类型。挂载参数参考[官方建议](https://cloud.google.com/compute/docs/disks/performance#optimize_local_ssd)增加了 `discard,nobarrier` 选项。完整挂载例子如下:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
GKE 当前的 NVMe 磁盘还在 Alpha 阶段，使用需特殊申请，不具备普遍意义。测试中，本地 SSD 盘统一使用 iSCSI 接口类型。挂载参数参考[官方建议](https://cloud.google.com/compute/docs/disks/performance#optimize_local_ssd)增加了 `discard,nobarrier` 选项。完整挂载例子如下：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+#### 网络
+
+GKE 网络模式使用具备更好扩展性以及功能强大的 [VPC-Native](https://cloud.google.com/kubernetes-engine/docs/how-to/alias-ips) 模式。在性能对比中，我们分别对 TiDB 使用 Kubernetes Pod 和 Host 网络分别做了测试。
+
+#### CPU
+
+在单可用集群测试中，我们为 TiDB/TiKV 选择 c2-standard-16 机型测试。
+在单可用于多可用集群对比测试中，因为 GCP 没有区域可以同时在 3 可用区里申请
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
在单可用与多可用集群的对比测试中，因为 GCP 没有区域可以同时在 3 个可用区里申请
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 10]
```diff
@@ -0,0 +1,407 @@
...
+c2-standard-16 机型。我们选择 n1-standard-16 机型测试。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
c2-standard-16 机型。我们选择了 n1-standard-16 机型测试。
```

Comment (lilin90) [reply dev/benchmark/sysbench-in-k8s.md]: Do not break lines manually unless you want to put them in different paras.

File: dev/benchmark/sysbench-in-k8s.md [hunk 11]
```diff
@@ -0,0 +1,407 @@
...
+
+### 操作系统及参数
+
+GKE 支持两种操作系统：COS 和 Ubuntu 。在 Point Select 测试中进行了对比测试。其余测试中，同一使用 Ubuntu 系统进行测试。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]: Do you mean "CentOS" by COS? If yes, let's make it accurate.

```suggestion
GKE 支持两种操作系统：CentOS 和 Ubuntu 。在 Point Select 测试中进行了对比测试。其余测试中，统一使用 Ubuntu 系统进行测试。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 12]
```diff
@@ -0,0 +1,407 @@
...
+
+内核统一做了如下配置:
+
+```shell
+sysctl net.core.somaxconn=32768
+sysctl vm.swappiness=0
+sysctl net.ipv4.tcp_syncookies=0
+```
+
+同时对最大文件数配置为 1000000 。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
同时对最大文件数配置为 1000000。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 13]
```diff
@@ -0,0 +1,407 @@
...
+
+### Sysbench 版本与运行参数
+
+本次测试中 sysbench 版本为 1.0.17 。并测试前统一使用 `oltp_common` 的 `prewarm` 命令进行数据预热。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
本次测试中 sysbench 版本为 1.0.17。并在测试前统一使用 `oltp_common` 的 `prewarm` 命令进行数据预热。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+  --db-driver=mysql \
+  --rand-type=uniform \
+  --rand-seed=$RANDOM \
+  --tables=16 \
+  --table-size=10000000 \
+  oltp_common \
+  prepare
+```
+
+`<tidb-host`> 为 tidb 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP, Service 域名，Host IP 以及 Load Balancer IP 。（下同）
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP。（下同）
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 15]
```diff
@@ -0,0 +1,407 @@
...
+  --db-driver=mysql \
+  --rand-type=uniform \
+  --rand-seed=$RANDOM \
+  --tables=16 \
+  --table-size=10000000 \
+  <test> \
+  run
+```
+
+`<test>` 为 sysbench 的测试 case 。我们选择了 oltp_point_select, oltp_update_index, oltp_update_no_index, oltp_read_write 这几种。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
`<test>` 为 sysbench 的测试 case。我们选择了 oltp_point_select、oltp_update_index、oltp_update_no_index、oltp_read_write 这几种。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 16]
```diff
@@ -0,0 +1,407 @@
...
+| Threads | QPS       | 95% latency(ms) |
+| ------- | --------- | --------------- |
+| 150     | 255981.11 | 1.06            |
+| 300     | 366482.22 | 1.50            |
+| 600     | 421279.84 | 2.71            |
+| 900     | 438730.81 | 3.96            |
+| 1200    | 441084.13 | 5.28            |
+| 1500    | 447659.15 | 6.67            |
+
+QPS 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
QPS 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 17]
```diff
@@ -0,0 +1,407 @@
...
+
+![Pod vs Host Network](/media/sysbench-in-k8s/pod-vs-host-network-qps.svg)
+
+Latency 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
Latency 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 18]
```diff
@@ -0,0 +1,407 @@
...
+
+![Pod vs Host Network](/media/sysbench-in-k8s/pod-vs-host-network-latency.svg)
+
+从图中可以看到 Host 网络下整体表现略好于 Pod 网络。
+
+#### Ubuntu vs COS
+
+GKE 平台可以为节点选择 [Ubuntu 和 COS 两种操作系统](https://cloud.google.com/kubernetes-engine/docs/concepts/node-images)。本次测试中，分别对两种操作系统进行 Point Select 测试。
+
+此次测试中 Pod 网络模式为 Host 。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
此次测试中 Pod 网络模式为 Host。
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 19]
```diff
@@ -0,0 +1,407 @@
...
+| Threads | QPS       | 95% latency(ms) |
+| ------- | --------- | --------------- |
+| 150     | 290690.51 | 0.74            |
+| 300     | 422941.17 | 1.10            |
+| 600     | 476663.44 | 2.14            |
+| 900     | 484405.99 | 3.25            |
+| 1200    | 489220.93 | 4.33            |
+| 1500    | 489988.97 | 5.47            |
+
+QPS 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
QPS 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 20]
```diff
@@ -0,0 +1,407 @@
...
+
+![COS vs Ubuntu](/media/sysbench-in-k8s/cos-vs-ubuntu-qps.svg)
+
+Latency 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
Latency 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 21]
```diff
@@ -0,0 +1,407 @@
...
+| Threads | QPS       | 95% latency(ms) |
+| ------- | --------- | --------------- |
+| 150     | 255981.11 | 1.06            |
+| 300     | 366482.22 | 1.50            |
+| 600     | 421279.84 | 2.71            |
+| 900     | 438730.81 | 3.96            |
+| 1200    | 441084.13 | 5.28            |
+| 1500    | 447659.15 | 6.67            |
+
+QPS 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
QPS 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 22]
```diff
@@ -0,0 +1,407 @@
...
+
+![Service vs Load Balancer](/media/sysbench-in-k8s/service-vs-load-balancer-qps.svg)
+
+Latency 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
Latency 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 23]
```diff
@@ -0,0 +1,407 @@
...
+| 1500    | 128501.34 | 344.082         |
+
+![OLTP Read Write](/media/sysbench-in-k8s/oltp-read-write-qps.svg)
+![OLTP Read Write](/media/sysbench-in-k8s/oltp-read-write-latency.svg)
+
+### 单可用区与多可用区对比
+
+GCP 多可用区，涉及跨 Zone 通信，网络延迟相比同 Zone 会少许增加。我们使用同样机器配置，对两种部署方案进行同一标准下的性能测试，了解多可用区延迟增加带来的影响。
+
+单可用区:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
单可用区：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 24]
```diff
@@ -0,0 +1,407 @@
...
+| Threads | QPS       | 95% latency(ms) |
+| ------- | --------- | --------------- |
+| 150     | 203879.49 | 1.37            |
+| 300     | 272175.71 | 2.30            |
+| 600     | 287805.13 | 4.10            |
+| 900     | 295871.31 | 6.21            |
+| 1200    | 294765.83 | 8.43            |
+| 1500    | 298619.31 | 10.27           |
+
+多可用区:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
多可用区：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 25]
```diff
@@ -0,0 +1,407 @@
...
+| Threads | QPS       | 95% latency(ms) |
+| ------- | --------- | --------------- |
+| 150     | 141027.10 | 1.93            |
+| 300     | 220205.85 | 2.91            |
+| 600     | 250464.34 | 5.47            |
+| 900     | 257717.41 | 7.70            |
+| 1200    | 258835.24 | 10.09           |
+| 1500    | 280114.00 | 12.75           |
+
+QPS 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
QPS 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 26]
```diff
@@ -0,0 +1,407 @@
...
+
+![Single Zonal vs Regional](/media/sysbench-in-k8s/single-zonal-vs-regional-qps.svg)
+
+Latency 对比:
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
Latency 对比：
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 27]
```diff
@@ -0,0 +1,407 @@
...
+
+![Single Zonal vs Regional](/media/sysbench-in-k8s/single-zonal-vs-regional-latency.svg)
+
+从图中可以看到在增大并发压力下，网络额外延迟产生的影响越来越小，额外的网络延迟将不是主要的性能瓶颈。
+
+## 结语
+
+此次测试主要将典型公有云部署 Kubernetes 运行 TiDB 集群的几种场景使用 sysbench 做了测试，了解不同因素可能带来的影响。从整体看，主要有以下几点：
+
+- VPC-Native 模式下 Host 网络性能略好于 Pod 网络 (~7%，以 QPS 差异估算，下同)
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
- VPC-Native 模式下 Host 网络性能略好于 Pod 网络（~7%，以 QPS 差异估算，下同）
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 28]
```diff
@@ -0,0 +1,407 @@
...
+- GCP 的 Ubuntu 系统单纯的读测试中性能略好于 COS (~9%)
+- 使用 Load Balancer 在集群外访问，会略损失性能 (~5%)
+- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响 (30% ~ 6%，随并发数增加而下降)
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
- 多可用区下节点之间延迟增加，会对 TiDB 性能产生一定的影响（30% ~ 6%，随并发数增加而下降）
```

File: dev/benchmark/sysbench-in-k8s.md [hunk 29]
```diff
@@ -0,0 +1,407 @@
...
+
+## 测试报告
+
+### 单可用区测试
+
+#### Pod Network vs Host Network
+
+Kubernetes 可以允许 Pod 运行在 Host 网络模式下，在 TiDB 实例独占机器没有端口冲突的情况下，此部署方式适用。我们分别对两种网络模式下分别做了 Point Select 测试。
+
+此次测试中，操作系统为 COS 。
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
此次测试中，操作系统为 COS。
```

**Comment by lilin90** [comment]
@cofyc And does this apply to `v3.0` too? If yes, please also add it to `v3.0`. Thanks!

File: dev/benchmark/sysbench-in-k8s.md (hunk 6 above)
Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: 是的

**Comment by lilin90** [comment]
Please add this new file to `TOC.md`.

**Comment by cofyc** [comment]
sure, I'll add when this is ready to merge

**Comment by cofyc** [comment]
Thanks for reviewing this! Sorry about style issues. 
I've added this file in `TOC.md` and fixed multiple lines issue in https://github.com/pingcap/docs-cn/pull/1803/commits/79e6862c02dd5cdd60258e62c8ffe870e4b5b160.

This applies to `3.0` too. I'll copy to `3.0` after @tennix LGTM this.

File: dev/benchmark/sysbench-in-k8s.md [hunk 30]
```diff
@@ -0,0 +1,405 @@
...
+```
+
+### 硬件配置
+
+#### 机器型号
+
+在单可用区测试中，我们选择了如下型号机器：
+
+| 组件     | 实例类型       | 数量  |
+| ----     | ----------     | ----- |
```
Comment (lilin90) [inline dev/benchmark/sysbench-in-k8s.md]: Please use left alignment for all table columns. Thanks!

**Comment by lilin90** [comment]
> Thanks for reviewing this! Sorry about style issues.
> I've added this file in `TOC.md` and fixed multiple lines issue in [79e6862](https://github.com/pingcap/docs-cn/commit/79e6862c02dd5cdd60258e62c8ffe870e4b5b160).
> 
> This applies to `3.0` too. I'll copy to `3.0` after @tennix LGTM this.

It's OK. I guess you must be familiar with the style rules now. ^_^

File: dev/benchmark/sysbench-in-k8s.md (hunk 30 above)
Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: done in https://github.com/pingcap/docs-cn/pull/1803/commits/0cb8281b91876783c0ec894cc323e22456f8c6ed, please check

File: dev/benchmark/sysbench-in-k8s.md [hunk 31]
```diff
@@ -0,0 +1,446 @@
...
+| 组件     | 实例类型       | 数量  |
+| :---     | :---------     | :---- |
+| PD       | n1-standard-4  | 3     |
+| TiKV     | n1-standard-16 | 3     |
+| TiDB     | n1-standard-16 | 3     |
+| Sysbench | n1-standard-16 | 3     |
+
+在高并发读测试中，压测端 sysbench 对 CPU 需求较高，需选择多核较高配置型号，避免压测端成为瓶颈。
+
+注：GCP 不同的区域可用机型不同。同时测试中发现磁盘性能表现也存在差异，我们统一在 us-central1 中申请机器测试。
```
Comment (aylei) [inline dev/benchmark/sysbench-in-k8s.md]: Use markdown quote?

File: dev/benchmark/sysbench-in-k8s.md [hunk 32]
```diff
@@ -0,0 +1,446 @@
...
+  --db-driver=mysql \
+  --rand-type=uniform \
+  --rand-seed=$RANDOM \
+  --tables=16 \
+  --table-size=10000000 \
+  oltp_common \
+  prepare
+```
+
+`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP。（下同）
```
Comment (aylei) [inline dev/benchmark/sysbench-in-k8s.md]:
```suggestion
`<tidb-host`> 为 TiDB 的数据库地址，根据不同测试需求选择不同的地址，比如 Pod IP、Service 域名、Host IP 以及 Load Balancer IP（下同）。
```

File: dev/benchmark/sysbench-in-k8s.md (hunk 31 above)
Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: done in https://github.com/pingcap/docs-cn/pull/1803/commits/a6b9cbc56d0f7a04c0b784b75db9fabcd54e38b3, please check

**Comment by lilin90** [comment]
@TomShawn Please note that these changes are added to v3.0 in #1809. And use `.png` not `.svg` for images.

**Comment by TomShawn** [comment]
> @TomShawn Please note that these changes are added to v3.0 in #1809. And use `.png` not `.svg` for images because [pdflatex does not support `.svg` format](https://github.com/pingcap/docs-cn/pull/1809#issue-312565385).

Sure!

File: dev/benchmark/sysbench-in-k8s.md [hunk 33]
```diff
@@ -0,0 +1,452 @@
...
+location-labels = ["region", "zone", "rack", "host"]
+```
+
+TiDB:
+
+```toml
+[log]
+level = "error"
+[prepared-plan-cache]
+enabled = true
```
Comment (DanielZhangQD) [inline dev/benchmark/sysbench-in-k8s.md]: max-procs is not set according to limits.cpu?

File: dev/benchmark/sysbench-in-k8s.md [hunk 34]
```diff
@@ -0,0 +1,452 @@
...
+在单可用集群测试中，我们为 TiDB/TiKV 选择 c2-standard-16 机型测试。在单可用与多可用集群的对比测试中，因为 GCP 没有区域可以同时在 3 个可用区里申请 c2-standard-16 机型。我们选择了 n1-standard-16 机型测试。
+
+### 操作系统及参数
+
+GKE 支持两种操作系统：COS (Container Optimized OS) 和 Ubuntu 。在 Point Select 测试中进行了对比测试。其余测试中，统一使用 Ubuntu 系统进行测试。
+
+内核统一做了如下配置:
+
+```shell
+sysctl net.core.somaxconn=32768
```
Comment (DanielZhangQD) [inline dev/benchmark/sysbench-in-k8s.md]: The net.core.somaxconn in the container is still 128 even if we set 32768 on the host, will this have any impact on the test?

Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: this sysctl parameter determines [backlog of the listen function](https://golang.org/src/net/sock_linux.go), however, we bench performance after all TCP connections are established, so it has no impact on our perf results

File: dev/benchmark/sysbench-in-k8s.md (hunk 33 above)
Comment (cofyc) [reply dev/benchmark/sysbench-in-k8s.md]: one tidb instance per node, so no need to set max-procs

**Pull request merged**